synapse/config/server.py

```diff
@@ -887,7 +888,7 @@ def read_gc_thresholds(
         )
 
 
-def parse_listener_def(num: int, listener: Any) -> ListenerConfig:
+def parse_listener_def(num: int, listener: JsonDict) -> ListenerConfig:
```

Comment:
I don't think we know for certain this is a `JsonDict` -- it can be anything that's valid YAML?

Reply:
👍 I've updated back to `Any` and added a `if not isinstance(listener, dict):` check so we still get type hints